tasks/review-app.js

```diff
@@ -1,208 +1,59 @@
-						? `, appId: ${app.id}`
-						: '';
-					throw new Error(`Review app not created yet. Current status: ${status}${appIdOutput}`);
-				};
-
-				return app.id;
-			});
-		return result;
-	};
-
-	return pRetry(checkForCreatedStatus, {
-		factor: RETRY_EXP_BACK_OFF_FACTOR,
-		retries: NUM_RETRIES,
-		minTimeout: RETRY_INTERVAL,
-		onFailedAttempt: (err) => {
-			const { attemptNumber, message } = err;
-			console.error(`${attemptNumber}/${NUM_RETRIES}: ${message}`); // eslint-disable-line no-console
-		}
-	});
-};
-
-const getAppName = async (appId) => {
-	const headers = await herokuHeaders();
-	return fetch(getAppUrl(appId), {
-		headers
-	})
-		.then(throwIfNotOk)
-		.then(res => res.json())
-		.then((result) => {
-			const { name } = result;
-			return name;
-		});
-};
-
-const deleteGitBranchApp = ({ pipelineId, branch, headers }) => {
-	const getReviewAppId = (pipelineId) => fetch(getPipelineReviewAppsUrl(pipelineId), {
-		headers
-	})
-		.then(throwIfNotOk)
-		.then(res => res.json())
-		.then((reviewApps = []) =>
-			reviewApps.find(
-				({ branch: reviewAppBranch }) => branch === reviewAppBranch)
-			)
-		.then((reviewApp) => {
-			const { id, app } = reviewApp;
-			if (!app) { throw new Error(`No app found in review app ${id}: ${JSON.stringify(reviewApp)}`); }
-
-			return app.id;
-		});
-	const deleteReviewApp = (appId) => fetch(getAppUrl(appId), {
-		headers,
-		method: 'delete'
-	}).then(throwIfNotOk);
-
-	return getReviewAppId(pipelineId)
-		.then(deleteReviewApp)
-		.catch(error => {
-			console.error(`Error in deleting review app: ${error}\nCarry on.`);  // eslint-disable-line no-console
-		});
-};
-
-async function task (app, options) {
+const {
+	createReviewApp,
+	findCreatedReviewApp,
+	waitTillReviewAppCreated,
+	waitForReviewAppBuild,
+	getAppName
+} = require('../lib/review-apps');
+
+async function task (appName, options) {
 	const { repoName, branch, commit, githubToken } = options;
+	const { id: pipelineId } = await pipelineInfo(appName);
 
-	const { id: pipelineId } = await pipelineInfo(app);
-	const headers = await herokuHeaders({ useReviewAppApi: true });
-	const body = {
-		pipeline: pipelineId,
-		branch,
-		source_blob: {
-			url: await getGithubArchiveRedirectUrl({ repoName, branch, githubToken }),
-			version: commit
-		}
-	};
-	const createReviewApp = () => fetch(REVIEW_APPS_URL, {
-		headers,
-		method: 'post',
-		body: JSON.stringify(body)
-	});
-
-	return createReviewApp()
+	return createReviewApp({ pipelineId, repoName, commit, branch, githubToken })
 		.then(res => {
 			const { status } = res;
 			if (status === 409) {
-				console.error(`Review app already created for ${branch} branch. Deleting existing review app first.`); // eslint-disable-line no-console
-				return deleteGitBranchApp({ pipelineId, branch, headers })
-					.then(createReviewApp);
+				console.error(`Review app already created for ${branch} branch. Using existing review app for build.`); // eslint-disable-line no-console
```

Comment:
Maybe this should be a warn or info rather than error?

Reply:
Nope, doesn't work with `warn` or `log` as it prints out to `stdin`, which I don't want since it needs to only print out the app name at the end.

eg, https://circleci.com/gh/Financial-Times/next-search-page/1922

The makefile thinks `TEST_APP=Review app already created for chores/nht-reuse-review-apps branch. Using existing review app for build. ft-next-sear-chores-nht-addywg` rather than `TEST_APP=ft-next-sear-chores-nht-addywg`

Reply:
Ah I see. Maybe worth adding comment saying why it needs to be an error.